File: androidApp/src/main/java/com/app/pustakam/android/screen/bookUIView/RichParagraphBlockView.kt
package com.app.pustakam.android.screen.bookUIView

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.app.pustakam.android.theme.PaperColor
import com.app.pustakam.android.theme.PaperInk
import com.app.pustakam.android.theme.typography
import com.app.pustakam.android.widgets.smartText.SmartTextColors
import com.app.pustakam.android.widgets.smartText.SmartTextDisplay
import com.app.pustakam.android.widgets.smartText.SmartTextTokens
import com.app.pustakam.core.filesys.reader.ReaderBlock

// 📖 15-Aug-2026: a formatted paragraph on a reader page — drawn by the editor's own renderer,
//   re-inked for paper so it stays legible whatever theme the app is in.
@Composable
fun RichParagraphBlockView(
    block: ReaderBlock.RichParagraph,
    modifier: Modifier = Modifier,
) {
    Column(modifier.fillMaxWidth()) {
        SmartTextDisplay(block = block.block, colors = readerTextColors())
        // continuation marker only when the paragraph had to be cut across pages
        if (block.chunkCount > 1) Text(
            "· ${block.chunkIndex} of ${block.chunkCount} ·",
            style = typography.labelSmall, color = PaperInk.copy(alpha = .5f),
            modifier = Modifier.align(Alignment.CenterHorizontally).padding(top = 6.dp),
        )
    }
}

@Composable
private fun readerTextColors(): SmartTextColors = SmartTextTokens.colors.copy(
    page = PaperColor,
    surface = PaperColor,
    onSurface = PaperInk,
    onSurfaceMuted = PaperInk.copy(alpha = .6f),
)

File: iosApp/iosApp/view/screens/bookReader/blocks/RichParagraphBlockView.swift
import SwiftUI
import shared

// 📖 15-Aug-2026: a formatted paragraph on a reader page — drawn by the editor's own renderer,
//   re-inked for paper so it stays legible whatever theme the app is in.
//   Mirrors Android RichParagraphBlockView.
struct RichParagraphBlockView: View {
    let block: ReaderBlock.RichParagraph

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            SmartTextDisplay(block: block.block, palette: Self.readerPalette)
            // continuation marker only when the paragraph had to be cut across pages
            if block.chunkCount > 1 {
                Text("· \(block.chunkIndex) of \(block.chunkCount) ·")
                    .font(.caption2)
                    .foregroundColor(BookPalette.ink.opacity(0.5))
                    .frame(maxWidth: .infinity)
            }
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }

    private static let readerPalette: SmartTextPalette = {
        let base = SmartTextPalette.light
        return SmartTextPalette(
            page: BookPalette.paper,
            surface: BookPalette.paper,
            toolbar: base.toolbar,
            onSurface: BookPalette.ink,
            onSurfaceMuted: BookPalette.ink.opacity(0.6),
            accent: BookPalette.cover,
            onAccent: BookPalette.paper,
            accentSoft: base.accentSoft,
            highlight: base.highlight,
            divider: BookPalette.ink.opacity(0.2),
            codeBackground: base.codeBackground,
            searchHit: base.searchHit,
            searchHitActive: base.searchHitActive
        )
    }()
}
